contracts/governance/FeeSharingCollectorProxy/FeeSharingCollector.sol

```diff
@@ -198,18 +195,42 @@ contract FeeSharingLogic is SafeMath96, IFeeSharingProxy, Ownable, FeeSharingPro
      * @param _amount Amount to be transferred.
      * */
     function transferTokens(address _token, uint96 _amount) public {
-        require(_token != address(0), "FeeSharingProxy::transferTokens: invalid address");
-        require(_amount > 0, "FeeSharingProxy::transferTokens: invalid amount");
+        require(
+            _token != ZERO_ADDRESS,
+            "FeeSharingCollectorProxy::transferTokens: invalid address"
+        );
+        require(_amount > 0, "FeeSharingCollectorProxy::transferTokens: invalid amount");
 
         /// @notice Transfer tokens from msg.sender
         bool success = IERC20(_token).transferFrom(address(msg.sender), address(this), _amount);
         require(success, "Staking::transferTokens: token transfer failed");
 
+        // if _token is wrbtc, need to unwrap it to rbtc
+        IWrbtcERC20 wrbtcToken = protocol.wrbtcToken();
+        if (_token == address(wrbtcToken)) {
+            wrbtcToken.withdraw(_amount);
+            _token = RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT;
+        }
```

Comment:
these changes have nothing to do with the story and the title. a PR to a branch containing the fee sharing updates would be easier to review. could you merge the most recent dev branch into the base branch here?

Reply:
merged the latest dev branch into this base branch